client/starwhale/api/_impl/service.py

```diff
@@ -0,0 +1,128 @@
+import json
+import base64
+import typing as t
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from starwhale.core.dataset.type import MIMEType, ArtifactType, BaseArtifact
+
+
+class Request(ABC):
+    @abstractmethod
+    def load(self, req: t.Any) -> t.Any:
+        raise NotImplementedError()
+
+
+class Response(ABC):
+    # TODO: support mime_type and body
+    @abstractmethod
+    def dump(self, resp: t.Any) -> bytes:
+        raise NotImplementedError()
+
+
+class GrayscaleImageRequest(Request):
```

Comment:
can we move the load function into [dataset/type.py](https://github.com/star-whale/starwhale/blob/main/client/starwhale/core/dataset/type.py) `GrayscaleImage` class?  From the user's view, the `GrayscaleImage`  can build a dataset and receive HTTP requests.

Reply:
```
from typing_extensions import Protocol
class Request(Protocol):
    def load(self, req: t.Any) -> t.Any:
         ...
class Response(Protocol):
    def dump(self, req: t.Any) -> t.Any:
         ...
```
can we use `Protocol` to define `Request` and `Response`?

Reply:
Done